processing/src/main/java/org/apache/druid/segment/StringDimensionMergerV9.java

```diff
@@ -71,6 +76,15 @@ public StringDimensionMergerV9(
   )
   {
     super(dimensionName, indexSpec, segmentWriteOutMedium, capabilities, progress, closer);
+    if (capabilities.hasSpatialIndexes()) {
+      Preconditions.checkArgument(
+          StringEncodingStrategy.UTF8.equals(indexSpec.getStringDictionaryEncoding().getType()),
+          StringUtils.format(
+              "Spatial indexes are incompatible with [%s] encoded dictionaries",
+              indexSpec.getStringDictionaryEncoding().getType()
+          )
+      );
```

Comment:
ah I can probably fix this, the reason it currently isn't hooked up is because it requires the index writer to implement the 'get' method for spatial indexes building step, `StringDImensionMergerV9.mergeIndexes`, which is the only caller. 

I haven't implemented it yet, currently the writer can only write, so it seemed easier to just put this check in for now. I can try to implement this method though so we can remove this limitation, either in this PR or a follow-up.

Reply:
i've fixed this by implementing the `get` method for the dictionary writer so this check has now been removed and `SpatialIndex` is now supported when using front-coding (also added tests)